File: longest-palindrome/longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        from collections import Counter
        if len(s) < 1:
            return 0
        
        s_dict = Counter(s)
        odd_longest, p_longest = 0, 0
        for key, val in s_dict.items():
            if val % 2 == 0: # Even number of vals
                p_longest += val
                
            elif val > odd_longest: # New largest odd num of vals
                p_longest += val
                if odd_longest > 0:
                    p_longest -= 1
                odd_longest = val
                
            else: # Get even num from odd num of vals
                p_longest += val - 1
                
        return p_longest

File: longest-palindrome/longest-palindrome.java
class Solution 
{
    public int longestPalindrome(String s) 
    {
        // Hash the provided string and count its characters
        HashMap<Character, Integer> sMap = new HashMap<>();
        for (char ch : s.toCharArray())
        {
            if (!sMap.containsKey(ch))
                sMap.put(ch, 1);
            else
                sMap.put(ch, sMap.get(ch)+1);
        }
        
        int length = 0;
        for (int value : sMap.values())
        {
            // Number of letters is even, just add to length
            if (value % 2 == 0)
                length += value;
            
            // Number of letters is odd & length is odd, add even number
            else if (length % 2 != 0)
                length += (value - 1);
            
            // Number of letters is odd & length is even, just add to length
            else
                length += value;
        }
        
        return length;
    }
}